include/support.h

```diff
@@ -132,6 +132,15 @@ char * safe_strcat(char (& dst)[N], const char * src) noexcept {
 	return & dst[0];
 }
 
+template<size_t N>
+int safe_strcmp(const char (& a)[N], const char * b) noexcept {
+	// Is 'b' at least as long as 'a'?
+	constexpr size_t N_without_null = N - 1;
+	if (strnlen(b, N) < N_without_null)
+		return -1;
+	return strncmp(a, b, N_without_null);
+}
+
```

Comment:
Yeah, I was just trying to programmatically get the needle-string length (instead of hardcoding), and also getting the string length of the haystack.   So even though `strnlen` is programmatically safe, it requires the developer to use it correctly - which is how it becomes unsafe in a "soft" sense. 

I will move it out of support and into the softmodem code; so other multi-char commands can use it.

Reply:
function moved.